onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -6402,17 +6402,29 @@ def aten_pinverse(self: TensorType, rcond: float = 1e-15) -> TensorType:
 
     raise NotImplementedError()
 
-
-def aten_pixel_shuffle(self: TensorType, upscale_factor: int) -> TensorType:
+@torch_op("aten::pixel_shuffle")
+def aten_pixel_shuffle(self: TReal, upscale_factor: int) -> TReal:
     """pixel_shuffle(Tensor self, int upscale_factor) -> Tensor"""
+    self_shape = op.Shape(self)
+    batch = self_shape[:-3]
+    C_out = op.Unsqueeze(self_shape[-3], [0])
+    H_out = op.Unsqueeze(self_shape[-2], [0])
+    W_out = op.Unsqueeze(self_shape[-1], [0])
+    # Reshaping input by collapsing all leading dimensions to match ONNX op requirement (4D)
+    reshaped_self = op.Reshape(self, op.Concat(op.Unsqueeze(-1, [0]), C_out, H_out, W_out, axis=0))
+    depth_to_space_output = op.DepthToSpace(reshaped_self, blocksize=upscale_factor, mode="CRD")
+    output_shape = op.Concat(batch, op.Shape(depth_to_space_output)[1:],axis=0)
+    return op.Reshape(depth_to_space_output, output_shape)
```

Comment:
Something looks odd with the above two lines ... they look like a no-op to me ... it changes just the first dim (at position 0) of the shape of the output tensor ... which must be a no-op for a Reshape.